encoder/vaapiencoder_h264.cpp

```diff
@@ -743,13 +743,21 @@ void VaapiEncoderH264::resetParams ()
     DEBUG("resetParams, ensureCodedBufferSize");
     ensureCodedBufferSize();
 
+    if (intraPeriod() == 0) {
+        ERROR("intra period must larger than 0");
+        assert(0);
```

Comment:
Here resetParams is a void function, if we change it to bool, upper-layer functions need to check it. And I think it's normal to assert parameters with ERROR info printed. We can see this way everywhere in  yami project and libva project.